primitives/storage/src/lib.rs

```diff
@@ -32,6 +34,67 @@ pub struct StorageKey(
 	pub Vec<u8>,
 );
 
+/// Storage key of a child trie, it contains the prefix to the key.
+#[derive(PartialEq, Eq, RuntimeDebug)]
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize, Hash, PartialOrd, Ord, Clone))]
+#[repr(transparent)]
+#[derive(RefCast)]
+pub struct PrefixedStorageKey(
+	#[cfg_attr(feature = "std", serde(with="impl_serde::serialize"))]
+	Vec<u8>,
+);
+
+impl Deref for PrefixedStorageKey {
+	type Target = Vec<u8>;
+
+	fn deref(&self) -> &Vec<u8> {
+		&self.0
+	}
+}
+
+impl DerefMut for PrefixedStorageKey {
+	fn deref_mut(&mut self) -> &mut Vec<u8> {
+		&mut self.0
+	}
+}
+
+impl PrefixedStorageKey {
+	/// Create a prefixed storage key from its byte array
+	/// representation.
+	/// Returns `None` on unknown prefix.
+	pub fn new(inner: Vec<u8>) -> Option<Self> {
```

Comment:
I don't know how much it worth checking that it is valid here even though a PrefixedStorageKey can be invalid.

Reply:
I think you are right, I am switching that.